### DISCUSSION TITLE
Release Notes on Major Releases

### DISCUSSION THREAD

**Discussion by mahyarmirrashed** [pallets/click#2770] [Q&A]
Could you list the specific breaking changes when upgrading the major version in their section? As a developer, it's hard to know what caused the bump from `6.x` to `7.x` and `7.x` to `8.x`.
As far as I can tell reading through the change log, it's just dropping support for older versions of Python?

**Comment by davidism** [comment]
You can find information about our version policy here: https://palletsprojects.com/versions. We do try to mark removals/renames/signature changes with deprecation warnings first, which are listed as "deprecate" then later "remove" in the change log. The only thing you can truly assume about version numbers is that they increase over time to indicate change; any change _could_ be breaking depending on how something was using it. Determining if you can upgrade is done by trying the upgrade, running tests, and addressing any failures.

**Reply by mahyarmirrashed** [reply]
I don't think that's totally fair to say. If we follow semantic versioning, this shouldn't be the case. API changes should be what's marked as breaking. All others, including dependencies being dropped, can be marked in a footnote because the TOML should reflect the current system dependencies. A Git blame or bisection could then figure out when certain dependency changes happened.
